enacts/calc.py

```diff
@@ -104,22 +107,67 @@ def synthesize_enacts(variable, time_res):
         # he rainy season is a bit shorter than 1/2 the year
         base_T = np.clip(base_T, a_min=0, a_max=None)
     # Coarse lat, lon dims to preserve some spatial homogeneity
-    Y = np.arange(2, 6.5, 0.5)
-    X = np.arange(-55, -51, 0.5)
-    # lat/lon log style trend around 1
-    # plus noise even closer to 1
-    lat_rand = np.log(Y*0.2/4.5 + 2.42) * (1 + 0.1 * np.random.randn(Y.size))
-    lon_rand = np.log(-X*0.2/4 - 0.05) * (1 + 0.1 * np.random.randn(X.size))
-    xrds = xr.Dataset(
-        {variable: (
-            ("X", "Y", "T"),
-            base_T * (lon_rand.reshape(-1, 1, 1) * lat_rand.reshape(1, -1, 1)),
-        )},
-        {"X": X, "Y": Y, "T": T},
-    # Interpolate back on a typical ENACTS spatial resolution
-    ).interp(X=np.arange(-55, -51, 0.0375), Y=np.arange(2, 6.5, 0.0375))
-    var_name = variable
-    return xrds[var_name]
+    Y = np.arange(bbox[1], bbox[3], 0.5)
+    X = np.arange(bbox[0], bbox[2], 0.5)
+    XY_rand = 0.1 * np.random.randn(X.size*Y.size).reshape(Y.size, X.size, 1)
```

Comment:
In my previous comment about putting Y before X, I should have mentioned that the convention more generally is to list variables from most slowly varying to most quickly varying. Most programming languages (Fortran is an exception) use [row-major order](https://en.wikipedia.org/wiki/Row-_and_column-major_order), which means that the row (X) varies more quickly than the column (Y), and that's why the order (Y, X) is the typical one.

Most datasets we deal with have T vary more slowly than both Y and X, so the order (T, Y, X) is the typical one. Here you're currently using (Y, X, T). Since the size of the T dimension here is 1, it makes no practical difference in this particular case, and xarray but we're probably better off following the convention anyway.

Reply:
Maybe, but I'm not aware of any C code that does matrix math.

In the last sentence of my previous comment, when I said "and xarray" I think I meant "and xarray will transpose arrays if necessary."